source/common/memory/utils.cc

```diff
@@ -22,18 +22,20 @@ void Utils::releaseFreeMemory() {
   so the overall impact on performance is small.
   Ref: https://github.com/envoyproxy/envoy/pull/9471#discussion_r363825985
 */
-void Utils::tryShrinkHeap() {
+void Utils::tryShrinkHeap(absl::optional<uint64_t> threshold) {
 #ifdef TCMALLOC
-  // TODO(zyfjeff): Make max unfreed memory byte configurable
-  static const uint64_t MAX_UNFREED_MEMORY_BYTE = 100 * 1024 * 1024;
+  constexpr uint64_t default_unfreed_memory = 100 * 1024 * 1024;
```

Comment:
this is the second mention of 100M in this PR -- should we make a constexpr in a header and share it?

Reply:
I think the other one is referring to the one in runtime::getInteger()
What do you think if we add a getOptionalInterger in runtime()?
Naturally the value in the runtime should be optional: who knows if there is such key. 

If you agree I can update the runtime interface of runtime